examples/cloud-operations/network-dashboard/main.tf

```diff
@@ -0,0 +1,183 @@
+resource "google_app_engine_application" "scheduler_app" {
+  project = module.project-monitoring.project_id
+  # "europe-west1" is called "europe-west" and "us-central1" is "us-central" for App Engine, see https://cloud.google.com/appengine/docs/locations
+  location_id = var.region == "europe-west1" || var.region == "us-central1" ? substr(var.region, 0, length(var.region) - 1) : var.region
+}
+
+# Create a storage bucket for the Cloud Function's code
+resource "google_storage_bucket" "bucket" {
+  name     = "net-quotas-bucket"
+  location = "EU"
+  project  = module.project-monitoring.project_id
+
+}
+
+data "archive_file" "file" {
+  type        = "zip"
+  source_dir  = "cloud-function"
+  output_path = "cloud-function.zip"
+  depends_on  = [google_storage_bucket.bucket]
+}
+
+resource "google_storage_bucket_object" "archive" {
+  # md5 hash in the bucket object name to redeploy the Cloud Function when the code is modified
+  name       = format("cloud-function#%s", data.archive_file.file.output_md5)
+  bucket     = google_storage_bucket.bucket.name
+  source     = "cloud-function.zip"
+  depends_on = [data.archive_file.file]
+}
+
+resource "google_cloudfunctions_function" "function_quotas" {
+  name        = "function-quotas"
+  project     = module.project-monitoring.project_id
+  region      = var.region
+  description = "Function which creates metric to show number, limit and utlizitation."
+  runtime     = "python39"
+
+  available_memory_mb   = 512
+  source_archive_bucket = google_storage_bucket.bucket.name
+  source_archive_object = google_storage_bucket_object.archive.name
+  service_account_email = module.service-account-function.email
+
+  timeout      = 180
+  entry_point  = "quotas"
+  trigger_http = true
+
+
+  environment_variables = {
+    monitored_projects_list = local.projects
+    monitoring_project_id   = module.project-monitoring.project_id
+
+    LIMIT_SUBNETS       = local.limit_subnets
+    LIMIT_INSTANCES     = local.limit_instances
+    LIMIT_INSTANCES_PPG = local.limit_instances_ppg
+    LIMIT_VPC_PEER      = local.limit_vpc_peer
+    LIMIT_L4            = local.limit_l4
+    LIMIT_L7            = local.limit_l7
+    LIMIT_L4_PPG        = local.limit_l4_ppg
+    LIMIT_L7_PPG        = local.limit_l7_ppg
+  }
+}
+
+resource "google_cloud_scheduler_job" "job" {
+  name        = "scheduler-net-dash"
+  project     = module.project-monitoring.project_id
+  region      = var.region
+  description = "Cloud Scheduler job to trigger the Networking Dashboard Cloud Function"
+  schedule    = var.schedule_cron
+
+  retry_config {
+    retry_count = 1
+  }
+
+  http_target {
+    http_method = "POST"
+    uri         = google_cloudfunctions_function.function_quotas.https_trigger_url
+    # We could pass useful data in the body later
+    body = base64encode("{\"foo\":\"bar\"}")
+  }
+}
+
+# TODO: How to secure Cloud Function invokation? Not   member = "allUsers" but specific Scheduler service Account?
```

Comment:
I think this is an important point. What exactly do you want to do with the SA?

Reply:
For now this is an HTTP trigger but it might make sense to have a pub/sub trigger instead and refactor that part. The goal is to ensure Cloud Scheduler can trigger the Cloud Function but also users are able to trigger it if required (for testing purposes).

Reply:
If you look at the quota monitor example we have, there's a pubsub trigger. I would avoid using HTTP at all costs.

Reply:
I changed it to a pub sub trigger.